### PR TITLE
"Implementing the exceptions provided."

### PR DESCRIPTION
**Pull request by MutebiAlexKato**
Review and suggest changes or approve for merging

### CODE REVIEW

File: src/org/pahappa/systems/exceptions/ValidationFailedException.java [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package org.pahappa.systems.exceptions;
 public class ValidationFailedException extends Exception {
 
-	/**
-	 * 
-	 */
+
 	private static final long serialVersionUID = 1L;
+	ValidationFailedException(String s){
```
Comment (katusiimeconrad) [inline src/org/pahappa/systems/exceptions/ValidationFailedException.java]: Maybe the parameter name can be more descriptive such as (String message)

**Review by katusiimeconrad — COMMENTED** [review]
Suggestion on the parameter name

File: src/org/pahappa/systems/exceptions/SavingFailedException.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
 package org.pahappa.systems.exceptions;
 public class SavingFailedException extends Exception {
 
-	/**
-	 * 
-	 */
+	public SavingFailedException(String message){
+		super("Incident not saved because" + message);
+	}
 	private static final long serialVersionUID = 1L;
```
Comment (Golder12) [inline src/org/pahappa/systems/exceptions/SavingFailedException.java:7]: usually better if this constant is before the method

Comment (Golder12) [inline src/org/pahappa/systems/exceptions/SavingFailedException.java]: usually better if the constant is before the method definition

**Review by ttcollins — APPROVED** [review]
Great work @MutebiAlexKato, please attend to the suggestions and changes requested. Also take note of the jira ticket ID.

**Review by NakagweSharifah — APPROVED** [review]

**Review by Golder12 — APPROVED** [review]

File: src/org/pahappa/systems/exceptions/SavingFailedException.java (hunk 2 above)
Comment (reifred) [reply src/org/pahappa/systems/exceptions/SavingFailedException.java]: @MutebiAlexKato kindly attend to this comment

File: src/org/pahappa/systems/exceptions/ValidationFailedException.java (hunk 1 above)
Comment (reifred) [reply src/org/pahappa/systems/exceptions/ValidationFailedException.java]: Attend to this comment

**Review by reifred — CHANGES_REQUESTED** [review]

**Comment by reifred** [comment]
> Great work @MutebiAlexKato, please attend to the suggestions and changes requested. Also take note of the jira ticket ID.

@MutebiAlexKato  pull request naming must follow what was agreed on.

File: src/org/pahappa/systems/exceptions/UpdateFailedException.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.pahappa.systems.exceptions;
+
+public class UpdateFailedException extends Exception{
+    UpdateFailedException(String message){
```
Comment (reifred) [inline src/org/pahappa/systems/exceptions/UpdateFailedException.java]: I don't think we need an extra UpdateFailedException class

**Review by reifred — CHANGES_REQUESTED** [review]

**Review by iandrew8 — APPROVED** [review]

**Review by katusiimeconrad — COMMENTED** [review]
Looks okay

File: src/org/pahappa/systems/exceptions/UpdateFailedException.java (hunk 3 above)
Comment (MutebiAlexKato) [reply src/org/pahappa/systems/exceptions/UpdateFailedException.java]: on it

**Review by reifred — APPROVED** [review]

**Pull request merged**